app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/PlanWizardInstancePropertiesStepReducer.js

```diff
@@ -90,9 +90,26 @@ export default (state = initialState, action) => {
         const bestGroup = { name: defaultSecurityGroupName, id: defaultSecurityGroupId };
 
         let preselectedFlavor;
+        let csvInvalidFlavorWarning = false;
         let preselectedGroup;
+        let csvInvalidGroupWarning = false;
 
-        if (editingPlan) {
+        if (existingInstancePropertiesRow.csvFields) {
+          const { osp_flavor, osp_security_group } = existingInstancePropertiesRow.csvFields;
+          if (osp_flavor) {
+            const matchingFlavor = tenantFlavors && tenantFlavors.find(flavor => flavor.name === osp_flavor);
+            preselectedFlavor = matchingFlavor && { name: matchingFlavor.name, id: matchingFlavor.id };
+            csvInvalidFlavorWarning = !matchingFlavor;
+          }
+          if (osp_security_group) {
+            const matchingGroup =
+              tenantSecurityGroups && tenantSecurityGroups.find(group => group.name === osp_security_group);
+            preselectedGroup = matchingGroup && { name: matchingGroup.name, id: matchingGroup.id };
+            csvInvalidGroupWarning = !matchingGroup;
+          }
+        }
+
+        if (editingPlan && !preselectedFlavor && !preselectedGroup) {
```

Comment:
Bit of an edge case here with regard to this conditional

1. Create a plan using CSV.  Specify security groups and flavors
2. Remove either the flavor column from the CSV, and invalidate one of the security group entries (possible use case:  user decides to edit plan to use best fit flavors, but still use predefined security groups.  Accidentally messes up a security group name during edit)
3. On the Instance Properties Step, the row with the invalid security group will correctly show the new warning, but will load the previously saved security group instead of falling back to the default, contrary to the warning messaging.